BreastCancer/Startup/APHAppDelegate.m

```diff
@@ -392,18 +474,393 @@ - (NSArray *)allSetTextBlocks
+                                                                        @"horizontalAccuracy",
+                                                                        @"horizontalAccuracyUnit",
+                                                                        @"verticalAccuracy",
+                                                                        @"verticalAccuracyUnit"];
+    APCPassiveDisplacementTrackingDataUploader* displacementSinker  = [[APCPassiveDisplacementTrackingDataUploader alloc]
+                                                                       initWithIdentifier:@"locationTracker"
+                                                                       columnNames:locationColumns
+                                                                       operationQueueName:@"APCDisplacement Tracker Sink"
+                                                                       andDataProcessor:nil];
+    [locationCollector setReceiver:displacementSinker];
+    [locationCollector setDelegate:displacementSinker];
+    [locationCollector start];
+    [self.passiveDataCollector addDataSink:locationCollector];
+}
+
+- (void)configureObserverQueries
+{
+    NSDate* (^LaunchDate)() = ^
+    {
+        APCUser*    user        = ((APCAppDelegate *)[UIApplication sharedApplication].delegate).dataSubstrate.currentUser;
+        NSDate*     consentDate = nil;
+        
+        if (user.consentSignatureDate)
+        {
+            consentDate = user.consentSignatureDate;
+        }
+        else
+{
+            NSFileManager*  fileManager = [NSFileManager defaultManager];
+            NSString*       filePath    = [[self applicationDocumentsDirectory] stringByAppendingPathComponent:@"db.sqlite"];
+            
+            if ([fileManager fileExistsAtPath:filePath])
+            {
+                NSError*        error       = nil;
+                NSDictionary*   attributes  = [fileManager attributesOfItemAtPath:filePath error:&error];
+                
+                if (error)
+                {
+                    APCLogError2(error);
+                    
+                    consentDate = [[NSDate date] startOfDay];
+                }
+                else
+                {
+                    consentDate = [attributes fileCreationDate];
+                }
+            }
+        }
+        
+        return consentDate;
+    };
+    
+    NSString*(^QuantityDataSerializer)(id) = ^NSString*(id dataSample)
+    {
+        HKQuantitySample*   qtySample           = (HKQuantitySample *)dataSample;
+        NSString*           startDateTimeStamp  = [qtySample.startDate toStringInISO8601Format];
+        NSString*           endDateTimeStamp    = [qtySample.endDate toStringInISO8601Format];
+        NSString*           healthKitType       = qtySample.quantityType.identifier;
+        NSString*           quantityValueRep    = [NSString stringWithFormat:@"%@", qtySample.quantity];
```

Comment:
Same comment applies here as in the other apps - should define a unit to use for each type, and use that unit, rather than using the user's unit preference.